### PR TITLE
delete task definition revisions and repository credentials

### PR DESCRIPTION
**Pull request by fishnix**
Deletes task definitions and any repository credentials in secrets manager.  This doesn't handle cluster deletion.

### CODE REVIEW

File: api/handlers_taskdef.go [hunk 1]
```diff
@@ -59,3 +60,49 @@ func (s *server) TaskDefCreateHandler(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+
+	output, err := orchestrator.DeleteTaskDef(r.Context(), &orchestration.TaskDefDeleteInput{
+		Cluster:        cluster,
+		TaskDefinition: taskdef,
+		Recursive:      recursive,
+	})
```
Comment (damnski) [inline api/handlers_taskdef.go:90]: like updating the struct as your call the function

**Comment by damnski** [comment]
very nice

File: api/handlers_services.go [hunk 2]
```diff
@@ -77,16 +77,6 @@ func (s *server) ServiceDeleteHandler(w http.ResponseWriter, r *http.Request) {
 	cluster := vars["cluster"]
 	service := vars["service"]
 
-	if cluster == "" {
-		handleError(w, apierror.New(apierror.ErrNotFound, "cluster cannot be empty", nil))
-		return
-	}
-
-	if service == "" {
```
Comment (tenyo) [inline api/handlers_services.go:85]: Why did you remove these checks, are they still needed?

**Review by tenyo — COMMENTED** [review]
Looks good 👍

File: api/handlers_services.go (hunk 2 above)
Comment (fishnix) [reply api/handlers_services.go:85]: It's not possible to get to the handler without those being set since they are defined in the routes
